priv/repo/migrations/20171106050209_add_pull_requests_to_task_list.exs

```diff
@@ -0,0 +1,53 @@
+defmodule CodeCorps.Repo.Migrations.AddPullRequestsToTaskList do
+  use Ecto.Migration
+
+  import Ecto.Query
+
+  alias CodeCorps.Repo
+
+  def up do
+    alter table(:task_lists) do
+      add :pull_requests, :boolean, default: false
+    end
+
+    flush()
+
+    # set all "In Progress" task lists to now contain pull requests
+    from(
+      tl in "task_lists",
+      where: [name: "In Progress"],
+      update: [set: [pull_requests: true]]
+    ) |> Repo.update_all([])
+
+    # get projects paired with associated pull request task list as ids
+    task_parent_data = from(
+      p in "projects",
+      left_join:
+        tl in "task_lists",
+        on: tl.project_id == p.id,
+        where: tl.pull_requests == true,
+      select: {p.id, tl.id}
+    ) |> Repo.all
+
+    # get all tasks for projects, associated to github pull requests and
+    # assign them to the pull request task list
+    task_parent_data |> Enum.each(fn {project_id, pr_list_id} ->
+      from(
+        t in "tasks",
+        where: [project_id: ^project_id],
+        where: not is_nil(t.github_issue_id),
+        inner_join:
+          gi in "github_issues",
+          on: t.github_issue_id == gi.id,
+          where: not is_nil(gi.github_pull_request_id),
+        update: [set: [task_list_id: ^pr_list_id]]
+      ) |> Repo.update_all([])
+    end)
```

Comment:
@JoshSmith I really feel this is the way to go with migrations.

If our schema modules do not match the schemas in any migration step, the code will fail, but if we go like this, "schemaless",  the only case where the migration step would not match the actual state of the schema were if we somehow run it manually, out of order, which would be incorrect usage of the migration.

Reply:
I really like this.

Reply:
Unfortunately it looks like this will override some of the work that was done in the earlier migration that moved tasks to closed, because this is not checking for whether the issues are closed.

Reply:
@JoshSmith You are correct about checking for gi not closed. You could also check for task not closed, it would likely serve the same purpose, since presumably, the task is otherwise in sync with the issue.

On a similar note, my other migration should probably add a clause for archived. I'm not sure what the exact query would be, but it would likely involve a `where: date_add(t.modified_at, 30, "day") > ^Date.utc_today` somewhere in it.

I can't push the change myself due to not being home, but that should help.